app/javascript/components/ItemCard.jsx

```diff
@@ -3,9 +3,57 @@ import Placeholder1 from "images/item_placeholder_1.JPG";
 import Placeholder2 from "images/item_placeholder_2.JPG";
 import EditIcon from "images/icons/edit_icon.svg";
 import DeleteIcon from "images/icons/delete_icon.svg";
+import { useCookies } from "react-cookie";
+
 
 export default(props) => {
 	const [onHover, setOnHover] = useState(false);
+	const [errorMessage, setErrorMessage] = useState(null);
+	const [cookies, _setCookie, _removeCookie] = useCookies(["ddough-auth"]);
+	const [userId] = useState();
+	
+
+	const buyNowHandler = async (e) => {
+        e.preventDefault();
+        setErrorMessage(null);
+
+        const body = {
+			doughnut_id: props.idx,
```

Comment:
`doughnut_id` and `idx` are not the same - the former is the doughnut's ID in the database and the latter is the position of the doughnut in the array of items that is stored in the frontend upon making the API call to get doughnuts. You'll want to pass down the doughnut ID from the item object and pass that in instead.